### DISCUSSION TITLE
React

### DISCUSSION THREAD

**Discussion by mark-whelan** [keplergl/kepler.gl#2696] [Q&A]
Does this work with the latest react versions? Done a bit looking around before i try this app and i'm finding lots of complaints that you need older react version for it to work.

**Comment by mark-whelan** [comment]
I also found lots of example here: https://codesandbox.io/examples/package/kepler.gl none of which work  :(

**Comment by mark-whelan** [comment]
I give up, this is virtually impossible to install on Windows.

**Reply by Anatolii-Ovcharuk** [reply]
To install a React app on Windows, you generally need Node.js and a package manager like npm (which comes with Node.js). Here’s a step-by-step guide:
1. Install Node.js and npm

    Download and install Node.js from the official site: Node.js Downloads.
    Choose the latest stable version and follow the installer instructions. This will install both Node.js and npm.
    Verify the installation by opening Command Prompt and running:

    node -v
    npm -v

2. Install Git (Optional, but Recommended)

    Git is commonly used for version control and may be needed if you’re cloning a React project from a repository.
    Download and install Git from git-scm.com.
    After installation, you can use git commands in Command Prompt or PowerShell.

3. Set Up Your React App

If you’re starting a new React project, use the create-react-app tool:

npx create-react-app my-app

    This command will create a new folder called my-app with a default React template.
    To navigate into the project folder, use:

    cd my-app

If you’re working with an existing project, download or clone the repository:

git clone <repository-url>
cd <project-folder>

4. Install Dependencies

    Inside the project folder, install the dependencies by running:

    npm install

5. Start the Development Server

    To start a local server and view the app in your browser, run:

    npm start

    The app should open in your default browser at http://localhost:3000.

Additional Tips

    Environment Variables: Some React apps require environment variables (like API keys). Create a .env file in the project root to store these, and add them in the format:

REACT_APP_API_KEY=your_api_key_here

Production Build: To build the app for production, use:

npm run build

This will generate optimized files in the build folder, ready for deployment.

**Comment by Anatolii-Ovcharuk** [comment]
Compatibility with the latest React versions can be an issue for certain libraries or applications, especially if they rely on older features or APIs that have since been deprecated or updated in newer React releases. If you’re seeing reports about compatibility problems, it could mean the app or some of its dependencies may not yet be fully adapted for the recent changes in React.

To check if this app will work with the latest React versions, you could:

    Review the Dependency List: Look at the package’s package.json file to see which React version is specified as compatible.

    Check for Maintainers' Notes: Often, maintainers will include information in the README or changelog on compatibility with the latest React versions or any specific versions that are recommended.

    Browse for Compatibility Fixes: If the app is open-source, look for issues or pull requests on the GitHub repository addressing compatibility updates. You may find community-created forks or patches that have resolved these issues.

If you are working on a larger project, you might consider testing the app in an isolated environment first. Alternatively, you could use tools like Docker or dependency managers to lock the app into a React version compatible with its current dependencies.
